### PR TITLE
Update error calculator for transducer

### PR DESCRIPTION
**Pull request by aky15**
Remove the padding when calculating the CER/WER for RNN-T in training.

### CODE REVIEW

File: espnet2/asr_transducer/error_calculator.py [hunk 1]
```diff
@@ -54,7 +54,10 @@ def __init__(
         self.report_wer = report_wer
 
     def __call__(
-        self, encoder_out: torch.Tensor, target: torch.Tensor
+        self,
+        encoder_out: torch.Tensor,
+        target: torch.Tensor,
+        encoder_out_lens: torch.Tensor,
     ) -> Tuple[Optional[float], Optional[float]]:
         """Calculate sentence-level WER or/and CER score for Transducer model.
```
Comment (b-flo) [inline espnet2/asr_transducer/error_calculator.py:62]: Could you also add the docstring for `encoder_out_lens` to the method, please?

i.e.: `encoder_out_lens: Encoder output sequences length. (B,)`

**Review by b-flo — COMMENTED** [review]
Hi,

Sorry about the delay, I missed the notification.

I agree with these changes, thanks! It wouldn't change a thing in terms of CER/WER but it may speed up the process a bit for batches with large padding.

**Comment by aky15** [comment]
Hi, @b-flo, I've added the docstring for encoder_out_lens to the method.
BTW, the modification will make the CER/WER calculation more accurate as the padding part can yield unexpected outputs.

**Comment by sw005320** [comment]
Thanks, @aky15.
It seems that the PR is ready, and once CI is passed, I'll merge this.

**Pull request merged**